src/modules/search.js

```diff
@@ -255,6 +257,105 @@ class Search {
+   * @param {number} [userParameters.clientId] - Client ID, utilized to personalize results
+   * @param {string} [userParameters.userId] - User ID, utilized to personalize results
+   * @param {string} [userParameters.segments] - User segments
+   * @param {object} [userParameters.testCells] - User test cells
+   * @param {string} [userParameters.userIp] - Origin user IP, from client
+   * @param {string} [userParameters.userAgent] - Origin user agent, from client
+   * @param {object} [networkParameters] - Parameters relevant to the network request
+   * @param {number} [networkParameters.timeout] - Request timeout (in milliseconds)
+   * @returns {Promise}
+   * @see https://docs.constructor.io/rest_api/search/natural_language_search/
+   * @example
+   * constructorio.search.getVoiceSearchResults('show me lipstick', {
+   *     resultsPerPage: 40,
+   * }, {
+   *     testCells: {
+   *         testName: 'cellName',
+   *    },
+   * });
+   */
+
+  getVoiceSearchResults(query, parameters = {}, userParameters = {}, networkParameters = {}) {
+    let requestUrl;
+    const { fetch } = this.options;
+    const controller = new AbortController();
+    const { signal } = controller;
+    const headers = {};
+
+    try {
+      const isVoiceSearch = true;
+      requestUrl = createSearchUrl(query, parameters, userParameters, this.options, isVoiceSearch);
+    } catch (e) {
+      return Promise.reject(e);
+    }
+
+    Object.assign(headers, helpers.combineCustomHeaders(this.options, networkParameters));
+
+    // Append security token as 'x-cnstrc-token' if available
+    if (this.options.securityToken && typeof this.options.securityToken === 'string') {
+      headers['x-cnstrc-token'] = this.options.securityToken;
+    }
+
+    // Append user IP as 'X-Forwarded-For' if available
+    if (userParameters.userIp && typeof userParameters.userIp === 'string') {
+      headers['X-Forwarded-For'] = userParameters.userIp;
+    }
+
+    // Append user agent as 'User-Agent' if available
+    if (userParameters.userAgent && typeof userParameters.userAgent === 'string') {
+      headers['User-Agent'] = userParameters.userAgent;
+    }
+
+    // Handle network timeout if specified
+    helpers.applyNetworkTimeout(this.options, networkParameters, controller);
+
+    return fetch(requestUrl, { headers, signal }).then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+
+      return helpers.throwHttpErrorFromResponse(new Error(), response);
+    }).then((json) => {
+      // Search results
+      if (json.response && json.response.results) {
+        if (json.result_id) {
+          json.response.results.forEach((result) => {
+            // eslint-disable-next-line no-param-reassign
+            result.result_id = json.result_id;
+          });
+        }
+
+        return json;
+      }
+
+      // Redirect rules
+      if (json.response && json.response.redirect) {
+        return json;
+      }
```

Comment:
Haven't had a chance to test this but wanted to ask if a response had a redirect field, is it guaranteed not to have a results field? If it does, then this section becomes unreachable.

Reply:
@Mudaafi  Yes if there is a redirect, the `results` field will be omitted and the response object would only have `redirect`